### PR TITLE
BACK-1670: Added business logic handlers to service discovery and updated discovery response format

### PR DESCRIPTION
**Pull request by sjl2024**

### CODE REVIEW

File: test/lib/dataLink.test.js [hunk 1]
```diff
@@ -115,13 +115,11 @@ describe('dataLink', () => {
...
       const testName = 'testObj';
-      const sampleTask = {};
       data.serviceObject(testName);
-      data.getServiceObjects(sampleTask, () => {
-        should.exist(sampleTask.serviceObjects[0]);
-        sampleTask.serviceObjects[0].should.eql(testName);
-        done();
-      });
+      const discoveredServiceObjects = data.getServiceObjects();
+      should.exist(discoveredServiceObjects[0]);
```
Comment (andrewmaillet) [inline test/lib/dataLink.test.js]: I haven't reviewed the rest of the tests, but can you verify that it's returning all of the objects here?

Comment (sjl2024) [inline test/lib/dataLink.test.js]: @andrewmaillet There's only one! The previous test calls `clearAll()` to remove any handlers, and this test only inserts a single handler into the store.

Comment (galniv) [inline test/lib/dataLink.test.js]: Right, but if there's only supposed to be one, it's a good idea to add `discoveredServiceObjects.length.should.equal 1` :)

File: lib/sdk.js [hunk 2]
```diff
@@ -37,8 +37,18 @@ class Service {
     const taskReceivedCallback = (task, completionCallback) => {
       if (task.taskType === 'dataLink') {
         return this.dataLink.process(task, this.moduleGenerator.generate(task), completionCallback);
-      } else if (task.taskType === 'dataLinkDiscovery') {
-        return this.dataLink.getServiceObjects(task, completionCallback);
+      } else if (task.taskType === 'serviceDiscovery') {
+        const discoveryObjects = {
```
Comment (galniv) [inline lib/sdk.js]: Why create the variable just to assign it right below and never use it again? you could just directly assign it to `task.discoveryObjects`

File: lib/service/businessLogic.js [hunk 3]
```diff
@@ -18,6 +18,10 @@ const notImplementedHandler = require('./notImplementedHandler');
 
 const registeredFunctions = new Map();
 
+function getHandlers() {
+  return [...registeredFunctions.keys()];
```
Comment (galniv) [inline lib/service/businessLogic.js]: Can you explain what `...registeredFunctions.keys()` does?

File: lib/sdk.js (hunk 2 above)
Comment (sjl2024) [inline lib/sdk.js]: Noted. Changing this.

File: lib/service/businessLogic.js (hunk 3 above)
Comment (sjl2024) [inline lib/service/businessLogic.js]: @galniv `registeredFunctions.keys()` is an es6 map function call which gets all of the keys in a `map` object. `...` is the es6 spread operator which essentially allows me to initialize a new array containing the `map.keys()` literals

Comment (galniv) [inline lib/service/businessLogic.js]: Interesting, thanks!

File: test/lib/businessLogic.test.js [hunk 4]
```diff
@@ -63,6 +67,21 @@ describe('business logic', () => {
...
+    afterEach((done) => {
+      logic.clearAll();
+      return done();
+    });
+    it('returns an array of all registered business logic handlers', (done) => {
+      const testHandlerName = 'testObj';
+      logic.register(testHandlerName, () => {});
+      const discoveredHandlers = logic.getHandlers();
+      should.exist(discoveredHandlers[0]);
+      (discoveredHandlers.length).should.eql(1);
```
Comment (galniv) [inline test/lib/businessLogic.test.js]: You don't need the parentheses around `discoveredHandlers.length`...

**Pull request merged**